pkg/api/pod/util.go

```diff
@@ -414,6 +415,13 @@ func DropDisabledTemplateFields(podTemplate, oldPodTemplate *api.PodTemplateSpec
 	dropDisabledFields(podSpec, podAnnotations, oldPodSpec, oldPodAnnotations)
 }
 
+// AddDisabledTemplateFieldsTo the provided set of resetFields
```

Comment:
I'm having trouble parsing the names of these functions
* what does it mean for a field to be "disabled"?
* why are eliding the name of the resource? I don't think the package name disambiguates sufficiently if that is the reason.

Reply:
"Disabled" means that a disabled feature gate leads to these fields being dropped from the object.

These apply to multiple resources (anything with a pod template here).

Maybe...
```
AddPotTemplateFeatureGatesDisabledFieldsTo
```
Now I feel like I'm writing Java. :-)

Reply:
1. A disabled field is a field that is related to a feature flag that is disabled on the API server, so even if you send a request with one of these fields set, the apiserver will drop it before persiting it.

2. I would guess that the reason is because these `AddDisabled*` functions are related to and thus should display similar ergonomics to the exisitng `DropDisabled*` functions. They are related in that the apiserver uses `DropDisabled*` to remove fields from the k8s object whild `AddDisabled*` adds fields to a set that gets used to remove them from a field managers set of managed fields.